Update getting started document (#149)
wso2/micro-integrator: examples/getting-started/README.md

 
2. Create an ESB solution project as explained [here](https://docs.wso2.com/display/EI6xx/Working+with+EI+Tooling#WorkingwithEITooling-CreatinganESBSolutionProjectCreatinganESBSolutionProject) and name it as 'compositeServiceProject'. Deselect the creation of a 'Registry Project' and a 'Connector Exporter Project' since we will not be needing them for this simple project. 
 
3. Open the project and [create a REST API](https://docs.wso2.com/display/EI6xx/Working+with+EI+Tooling#WorkingwithEITooling-CreatingartifactsforanESBSolutionProject). You can either create an API from scratch and configure it as shown in the following diagram  or directly import the config from [here](composite-service/composite-service/src/main/synapse-config/api/forwardOrderApi.xml).
 ![diagram](images/api-config.png) 
 
4. Save the project and export the composite application to a desired location. (You can refere [How to working with EI Tooling](https://docs.wso2.com/display/EI6xx/Working+with+EI+Tooling#WorkingwithEITooling-PackagingESBartifacts) document for get to know more information about this)

> **Note -** Alternatively you can build the project 'composite-service' using maven and 'composite-serviceCompositeApplication_1.0.0.car' C-App will be located in `<mi-work-directory>/examples/getting-started/composite-service/composite-service-capp/target` when you use the following command.

Reviewer: ```suggestion
**Note -** Alternatively, you can build the project 'composite-service' using maven and 'composite-serviceCompositeApplication_1.0.0.car' C-App will be located in `<mi-work-directory>/examples/getting-started/composite-service/composite-service-capp/target` when you use the following command.
```

Author: Done

```
   $ cd <mi-work-directory>/examples/getting-started/composite-service
   $ mvn clean install
   ```
## Deployment

> Through out this section, it will be assumed that you have created two C-apps for 'composite-service' and 'order-process-backend' which are reside in ```<mi-work-directory>/examples/getting-started/composite-service/composite-service-capp/target``` and ```<mi-work-directory>/examples/getting-started/order-process-be/order-processCompositeApplication/target``` folders.

Reviewer: ```suggestion
Throughout this section, it will be assumed that you have created two C-apps for 'composite-service' and 'order-process-backend' which are reside in ```<mi-work-directory>/examples/getting-started/composite-service/composite-service-capp/target``` and ```<mi-work-directory>/examples/getting-started/order-process-be/order-processCompositeApplication/target``` folders.
```

Author: Done

   ```
#### Deploying the composite service locally

Assume you have followed steps 1-4 in the section [Create the composite service to invoke the back-end.](#2-create-the-composite-service-to-invoke-the-back-end)

1. Copy the `composite-service/composite-service-capp/target/composite-serviceCompositeApplication_1.0.0.car` into `<MI2_HOME>/repository/deployment/server/carbonapps`

2. Start the Micro Integrator instance(MI2).

You have successfully deployed 'order-process-backend' and the 'composite-service' services locally. Now you can test your deployments by using [Testing](#Testing) section.

### Docker Deployment
Please note that the distribution must be built with ```-Ddocker.skip=false``` prior to this step, for the base image to be created. And also assume that you have followed all the steps in the section [Create the order processing back-end.](#1-create-the-order-processing-back-end) and [Create the composite service to invoke the back-end.](#2-create-the-composite-service-to-invoke-the-back-end)

> In [Create the composite service to invoke the back-end](#2-create-the-composite-service-to-invoke-the-back-end) step 3 you need to replace 'order-process-backend' endpoint address as follows because in docker containers each container has their own localhost.

Reviewer 2: Shall we use a link to point to the file https://github.com/wso2/micro-integrator/blob/master/examples/getting-started/composite-service/composite-service/src/main/synapse-config/api/forwardOrderApi.xml?

Author: Done

   **Note -** If you want to start Micro-Integrator with a portOffset=1 you have to pass the parameter as ```-DportOffset=11```, because Micro-Integrator starts with a default portOffset=10.
      
3. You should be able to access the back-end using the below curl command
   ```
   curl --header "Content-Type: application/json" --request POST  --data '{"store": {"book": [{"author": "Nigel Rees","title": "Sayings of the Century"},{"author": "J. R. R. Tolkien","title": "The Lord of the Rings","isbn": "0-395-19395-8"}]}}' http://localhost:8291/order
   ```
##### Deploying the composite service locally

Assume you have followed steps 1-4 in the section [Create the composite service to invoke the back-end.](#2-create-the-composite-service-to-invoke-the-back-end)

1. Copy the `composite-service/composite-service-capp/target/composite-serviceCompositeApplication_1.0.0.car` into `<MI2_HOME>/repository/deployment/server/carbonapps`

2. Start the Micro Integrator instance(MI2).

> You have deployed 'order-process-backend' and the 'composite-service' services successfully in locally. Now you can test your deployments by using [Testing](#Testing) section.

Reviewer: ```suggestion
You have successfully deployed 'order-process-backend' and the 'composite-service' services locally. Now you can test your deployments by using [Testing](#Testing) section.
```

Author: Done
